shared/src/main/java/uk/gov/di/authentication/shared/services/NoSessionOrchestrationService.java

```diff
@@ -0,0 +1,101 @@
+        this.redisConnectionService = redisConnectionService;
+        this.clientSessionService = clientSessionService;
+        this.configurationService = configurationService;
+    }
+
+    public NoSessionEntity generateNoSessionOrchestrationEntity(
+            Map<String, String> queryStringParameters, boolean noSessionResponseEnabled)
+            throws NoSessionException {
+        LOG.info(
+                "Attempting to generate error response using state. NoSessionResponseEnabled: {}",
+                noSessionResponseEnabled);
+        if (isAccessDeniedErrorAndStatePresent(queryStringParameters, noSessionResponseEnabled)) {
+            LOG.info("access_denied error and state param are both present");
+            var clientSessionId =
+                    getClientSessionIdFromState(State.parse(queryStringParameters.get("state")))
+                            .orElseThrow(
+                                    () ->
+                                            new NoSessionException(
+                                                    "ClientSessionId could not be found using state param"));
+            LOG.info("ClientSessionID found using state");
+            attachLogFieldToLogs(CLIENT_SESSION_ID, clientSessionId);
+            attachLogFieldToLogs(GOVUK_SIGNIN_JOURNEY_ID, clientSessionId);
+            var clientSession =
+                    clientSessionService
+                            .getClientSession(clientSessionId)
+                            .orElseThrow(
+                                    () ->
+                                            new NoSessionException(
+                                                    "No client session found with given client sessionId"));
+            LOG.info("ClientSession found using clientSessionId");
+            var errorDescription =
+                    Optional.ofNullable(queryStringParameters.get("error_description"))
+                            .orElse(OAuth2Error.ACCESS_DENIED.getDescription());
+            var errorObject = new ErrorObject(queryStringParameters.get("error"), errorDescription);
+            LOG.info(
+                    "ErrorObject created for session cookie not present. Generating NoSessionEntity in preparation for response to RP");
+            return new NoSessionEntity(clientSessionId, errorObject, clientSession);
+        } else {
+            LOG.warn(
+                    "Session Cookie not present and access_denied or state param missing from error response. NoSessionResponseEnabled: {}",
+                    noSessionResponseEnabled);
+            throw new NoSessionException(
+                    format(
+                            "Session Cookie not present and access_denied or state param missing from error response. NoSessionResponseEnabled: %s",
+                            noSessionResponseEnabled));
+        }
+    }
+
+    public void storeClientSessionIdAgainstState(String clientSessionId, State state) {
```

Comment:
When will this get called?

Reply:
It will get called during either the IPV or DCMAW handoff